### PR TITLE
Adds a global var finder. Makes find_references search globals

### PR DESCRIPTION
**Pull request by Cyberboss**
```
## TESTING: Beginning search for references to a /obj/effect/landmark/start.
## TESTING: Found /obj/effect/landmark/start [start*wizard] in global list start_landmarks_list.
```

DMTreeToGlobalsList is pretty simple. You feed it the output of dm.exe -o and it lists out all global variables with optional prefix and postfix

### CODE REVIEW

**Comment by RemieRichards** [comment]
don't rely on parsing the XML, it's broken in a lot of places and no longer a supported feature.

**Comment by Cyberboss** [comment]
It works for this case

**Comment by RemieRichards** [comment]
for now.

File: code/controllers/subsystem/garbage.dm [hunk 1]
```diff
@@ -326,4 +327,735 @@ var/datum/subsystem/garbage_collector/SSgarbage
...
+				var/variable = D.vars[varname]
+				if(variable == src)
+					testing("Found [src.type] \ref[src] in [D.type]'s [varname] var. Global: [Xname]")
+				else if(islist(variable) && src in variable)
+					testing("Found [src.type] \ref[src] in [D.type]'s [varname] list var. Global: [Xname]")
+	CHECK_TICK
+
+//if find_references isn't working for some datum
+//update this list using tools/DMTreeToGlobalsList
+/datum/proc/find_references_in_globals()
```
Comment (optimumtact) [inline code/controllers/subsystem/garbage.dm]: jeeeeeeesus chriiiiiiiiiiiiiiiiiiist

Comment (Cyberboss) [inline code/controllers/subsystem/garbage.dm]: Yeah, code/_globalvars/ is a fucking meme

**Pull request merged**